### PR TITLE
Implement GOP module

### PR DESCRIPTION
**Pull request by 0xKaito**

### CODE REVIEW

File: sources/baseNFT.move [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        name: String,
+        /// Description of the token
+        description: String,
+    }
+
+    // ===== Entrypoints =====
+
+    /// Module initializer is called only once on module publish.
+    fun init(ctx: &mut TxContext) {
+
```
Comment (rishabh0x00) [inline sources/baseNFT.move]: remove this space

File: sources/gop.move [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        claimed: bool,
+        airdrop: bool,
+        ieo: bool
+    }
+
+    struct AdminCap has key {
+        id: UID
+    }
+
+    struct GopCap has key {
```
Comment (rishabh0x00) [inline sources/gop.move]: Why do we need this?

File: sources/gop.move [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        id: UID
+    }
+
+    // ===== Events =====
+
+    struct GopUpdated has copy, drop {
```
Comment (rishabh0x00) [inline sources/gop.move]: GOPAttributesUpdated

File: sources/gop.move [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        claimed: bool,
+        airdrop: bool,
+        ieo: bool
+    }
+
+    /// One-Time-Witness for the module.
+    struct GOP has drop {}
```
Comment (rishabh0x00) [inline sources/gop.move:54]: Please have consistency with "GOP" and "Gop", lets use GOP instead of Gop everywhere.

File: sources/gop.move [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+                airdrop: false,
+                ieo: false
+            });
+
+            index = index + 1;
+        }
+    }
+
+    /// Permanently delete `GOP`
+    public entry fun burn<T: key + store>(nft: baseNFT::NFT<T>, ctx: &mut TxContext) {
```
Comment (rishabh0x00) [inline sources/gop.move]: can be moved to base

File: sources/gop.move [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+        baseNFT::burn(nft, ctx);
+    }
+
+    /// Transfer `GOP` to `recipient`
+    public entry fun transfer_nft<T: key + store>(
```
Comment (rishabh0x00) [inline sources/gop.move]: to base

File: sources/gop.move [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+        nft: baseNFT::NFT<T>, recipient: address, ctx: &mut TxContext
+    ) {
+        baseNFT::transfer_nft(nft, recipient, ctx);
+    }
+
+    // === AdminCap Functions ===
+
+    /// Update the metadata of `GOP`
+    public entry fun update_metadata(
```
Comment (rishabh0x00) [inline sources/gop.move]: If this is visible from both base and GOP modules, lets only have one is base

File: sources/gop.move [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+
+    /// transfer publisher object to new_owner
+    public entry fun transfer_publisher_object(_: &AdminCap, publisher_object: package::Publisher ,new_owner: address, _: &mut TxContext) {
+        transfer::public_transfer(publisher_object, new_owner);
+    }
+
+    /// transfer Upgrade to new_owner
+    public entry fun transfer_upgrade_cap(_: &AdminCap, upgradeCap: package::UpgradeCap ,new_owner: address, _: &mut TxContext) {
+        transfer::public_transfer(upgradeCap, new_owner);
+    }
```
Comment (rishabh0x00) [inline sources/gop.move:185]: If we have admin cap in base, we can move this to base as well

Comment (0xKaito) [reply sources/gop.move:185]: we are using two types of admin, base admin will different from baseNFT admin

**Pull request closed without merging**